src/main/java/org/ow2/proactive/connector/iaas/cloud/provider/jclouds/JCloudsProvider.java

```diff
@@ -146,7 +153,10 @@ public void deleteInstance(Infrastructure infrastructure, String instanceId) {
         try {
             execResponses = getComputeServiceFromInfastructure(infrastructure).runScriptOnNodesMatching(runningInGroup(instanceTag),
                                                                                                         buildScriptToExecuteString(instanceScript),
-                                                                                                        buildScriptOptions(instanceScript));
+                                                                                                        buildScriptOptions(instanceScript,
+                                                                                                                           null,
```

Comment:
why null as second parameter?

Reply:
Regarding your two comments, I propose that I create a specific `buildScriptOptions` method for each case which will be more explicit, like: `buildScriptOptionsWithInstanceId` and `buildScriptOptionsWithInstancetag`. Deal?